### DISCUSSION TITLE
Introduce Externald

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#6194] [Ideas]
I feel like it would be useful to introduce a field that would be on all object, similar to position, called "externalId". This would help a lot when syncing with external systems and we could hook some useful logic linked to that (enforce unicity constraint etc)

**Comment by FelixMalfait** [comment]
@thomtrp maybe that could be useful for remote objects too?

Just saw Salesforce has that concept already
https://help.salesforce.com/s/articleView?id=sf.xorg_external_object_record_id.htm&type=5
https://www.saasguru.co/external-id-in-salesforce/

**Comment by thomtrp** [comment]
@FelixMalfait it would be useful if we wanted to store data on our side associated to the remotes. But since we do not want to store anything, not sure how we should use it?

**Comment by FelixMalfait** [comment]
Moving to discussions as I'm not sure this is relevant.
